### PR TITLE
fix namespace of extensions in ssl.h

### LINKED ISSUE #17: Fix namespacing of KE and Extension identifiers in ssl.h
`ssl.h` contains identifiers outside the `MBEDTLS_` namespace, e.g.
```c
#define NO_EXTENSION 0
#define PRE_SHARED_KEY_EXTENSION 1
#define KEY_SHARE_EXTENSION 2
#define SIGNATURE_ALGORITHM_EXTENSION 4
#define SUPPORTED_GROUPS_EXTENSION 8
#define MAX_FRAGMENT_LENGTH_EXTENSION 16
#define ALPN_EXTENSION 32
#define SUPPORTED_VERSION_EXTENSION 64
#define PSK_KEY_EXCHANGE_MODES_EXTENSION 128
#define EARLY_DATA_EXTENSION 256
#define SERVERNAME_EXTENSION 512
#define COOKIE_EXTENSION 1024
#define CID_EXTENSION 2048
```

Go through `ssl.h` and adjust the namespacing so everything is within `MBEDTLS_`.

### PR DESCRIPTION
**Pull request by yuhaoth**
## Description
This patch is to fix  #17

## Status
**READY**

## Migrations
 NO

## Todos
- [X] Tests
   - Test pass with `ssl-opt-in-docker.sh`

### CODE REVIEW

File: include/mbedtls/ssl.h [hunk 1]
```diff
@@ -167,19 +167,19 @@
...
-#define SUPPORTED_GROUPS_EXTENSION 8
-#define MAX_FRAGMENT_LENGTH_EXTENSION 16
-#define ALPN_EXTENSION 32
-#define SUPPORTED_VERSION_EXTENSION 64
-#define PSK_KEY_EXCHANGE_MODES_EXTENSION 128
-#define EARLY_DATA_EXTENSION 256
-#define SERVERNAME_EXTENSION 512
-#define COOKIE_EXTENSION 1024
-#define CID_EXTENSION 2048
+#define MBEDTLS_NO_EXTENSION                        0
```
Comment (hanno-becker) [inline include/mbedtls/ssl.h]: We need a more refined namespace here, top-level `MBEDTLS_` is too broad.

What about `MBEDTLS_SSL_EXT_ABC`

**Review by hanno-becker — CHANGES_REQUESTED** [review]
We need a lower-level namespace than `MBEDTLS_`, see comment.

File: include/mbedtls/ssl.h (hunk 1 above)
Comment (yuhaoth) [reply include/mbedtls/ssl.h]: Understand

**Review by hanno-becker — APPROVED** [review]
LGTM

**Pull request merged**